File: testing/mobile/utils/excel_report.py
import openpyxl
from openpyxl.styles import PatternFill
import os
import datetime

def create_excel_report(test_results, filename='report_mobile.xlsx'):
    wb = openpyxl.Workbook()
    ws = wb.active
    ws.title = "Test Results"

    # Define headers
    headers = ['Test ID', 'Category', 'Test Case Name', 'Status', 'Error/Details', 'Timestamp']
    ws.append(headers)

    # Styles
    pass_fill = PatternFill(start_color="90EE90", end_color="90EE90", fill_type="solid")
    fail_fill = PatternFill(start_color="FFCCCB", end_color="FFCCCB", fill_type="solid")

    for result in test_results:
        row = [
            result.get('id'),
            result.get('category'),
            result.get('name'),
            result.get('status'),
            result.get('details'),
            result.get('timestamp')
        ]
        ws.append(row)
        
        # Color coding
        last_row = ws.max_row
        status_cell = ws.cell(row=last_row, column=4) # Status is 4th col
        if result.get('status') == 'Pass':
            status_cell.fill = pass_fill
        else:
            status_cell.fill = fail_fill

    # Save
    save_path = os.path.join(os.path.dirname(__file__), '..', '..', filename)
    wb.save(save_path)
    print(f"Mobile Report generated: {save_path}")

File: selenium_testing/utils/excelReport.js
const ExcelJS = require('exceljs');
const path = require('path');

async function createExcelReport(testResults, filename = 'report_web.xlsx') {
    const workbook = new ExcelJS.Workbook();
    const worksheet = workbook.addWorksheet('Test Results');

    worksheet.columns = [
        { header: 'Test ID', key: 'id', width: 10 },
        { header: 'Category', key: 'category', width: 20 },
        { header: 'Test Case Name', key: 'name', width: 40 },
        { header: 'Status', key: 'status', width: 15 },
        { header: 'Error/Details', key: 'details', width: 50 },
        { header: 'Timestamp', key: 'timestamp', width: 25 }
    ];

    testResults.forEach(result => {
        const row = worksheet.addRow(result);
        const statusCell = row.getCell('status');
        if (result.status === 'Pass') {
            statusCell.fill = { type: 'pattern', pattern: 'solid', fgColor: { argb: 'FF90EE90' } };
        } else {
            statusCell.fill = { type: 'pattern', pattern: 'solid', fgColor: { argb: 'FFFFCCCB' } };
        }
    });

    const filePath = path.join(__dirname, '..', filename);
    await workbook.xlsx.writeFile(filePath);
    console.log(`Web Report generated: ${filePath}`);
}

module.exports = { createExcelReport };
